Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5492,6 +5492,7 @@ void RenderLayer::updateFilterPaintingStrategy()
     }
     
     ensureLayerFilters();
+    renderer().view().setHasSoftwareFilters(!renderer().page().acceleratedFiltersEnabled());
```

Comment:
It's not clear why this fixes any repaint bugs; it's not really the correct approach.

The right approach is something more complex, which is to teach `RenderBlock::computeOverflow()` about how to add visual overflow if some ancestor has a drop-shadow filter.